### ISSUE TITLE
Delete unmaintained branches

### ISSUE THREAD

**Issue by rootulp** [celestiaorg/celestia-core#830]
It's difficult to see which branches are actively maintained on this repo. If we delete unmaintained branches in Github UI, then we can enable contributors to more easily see the branches that are relevant, (likely a handful including `v0.34.x-celestia` and `v0.35.x-celestia`). I don't think this is risky b/c it's possible to restore a deleted branch from Github UI. 

<img width="317" alt="Screen Shot 2022-08-12 at 3 17 50 PM" src="https://user-images.githubusercontent.com/3699047/184428300-42ebe9c7-3200-4caa-833d-459b0db0fd5f.png">

**Comment by rootulp** [comment]
@evan-forbes any objections to me doing this for branches that haven't been updated in the past month?

**Comment by rootulp** [comment]
To reduce the frequency of this chore, a repo admin should enable this setting on https://github.com/celestiaorg/celestia-core/settings:


<img width="790" alt="Screen Shot 2022-08-14 at 11 59 05 AM" src="https://user-images.githubusercontent.com/3699047/184545126-df7d8307-8c93-4744-a4be-22ccabfaee7d.png">

**Comment by evan-forbes** [comment]
>To reduce the frequency of this chore, a repo admin should enable this setting on https://github.com/celestiaorg/celestia-core/settings:

this is already set :grimacing: I think I just need to stop polluting origin :sweat_smile: 

I'll move all of my old branches over to a fork later today

**Comment by rootulp** [comment]
Sounds good, please assign back to me after you move branches and I can delete the remainder of stale branches

**Comment by rootulp** [comment]
We also want to deprecate `master` to avoid links to outdated content. Example: https://github.com/celestiaorg/docs/blob/069623334073d5b2c4b626f732f016c2ede18841/docs/concepts/how-celestia-works/data-availability-layer.md?plain=1#L152

**Comment by evan-forbes** [comment]
apologies again for the delay on this, but I've removed all of my branches. There are still a few old ones that are not mine, but I don't feel comfortable deleting other people's branches.

some of them were quite nostalgic. I almost just want to keep them forever as grafiti somehwere lol

**Comment by rootulp** [comment]
> apologies again for the delay on this

No worries on the delay, this is a low priority chore. 

> I don't feel comfortable deleting other people's branches.

It seems possible to restore deleted branches so I'm comfortable doing this unless you object. 

Ref: https://stackoverflow.com/questions/4674226/does-github-keep-deleted-remote-branches-in-history-if-so-can-those-be-restore

**Comment by rootulp** [comment]
<img width="367" alt="Screen Shot 2022-09-14 at 10 28 57 AM" src="https://user-images.githubusercontent.com/3699047/190182797-5a848ff7-d64c-4eb7-b994-4c92dfd74c8c.png">

master has a branch protection rule that prevents me from deleting it

**Comment by rootulp** [comment]
I deleted branches that haven't been touched in over 6 months: https://github.com/celestiaorg/celestia-core/branches

We're down to 10 branches.
<img width="323" alt="Screen Shot 2022-09-22 at 10 50 08 AM" src="https://user-images.githubusercontent.com/3699047/191779782-cd702181-f5ba-4713-874a-2265e19ec299.png">

Tagging @evan-forbes if you have admin rights to delete the `master` branch.

**Comment by evan-forbes** [comment]
I'm opposed to deleting the master branch, as it has a lot of history for the work we did before we started the new approach of treating tendermint as a black box. Its likely not that useful, but perhaps we'll want to remember some weird thing that we did.

**Comment by rootulp** [comment]
If it's strictly for reference purposes, are you open to renaming `master` to `deprecated-master` in order to avoid confusion like https://github.com/celestiaorg/celestia-core/pull/795#pullrequestreview-1030657928

**Comment by evan-forbes** [comment]
updated
